src/ServiceControl/Operations/BodyStorage/BodyStorageFeature.cs

```diff
@@ -58,36 +58,39 @@ static string GetContentType(IReadOnlyDictionary<string, string> headers, string
                 return contentType;
             }
 
-            async Task StoreBody(byte[] body, IReadOnlyDictionary<string, string> headers, IDictionary<string, object> metadata, int bodySize, string contentType)
+            async ValueTask StoreBody(byte[] body, IReadOnlyDictionary<string, string> headers, IDictionary<string, object> metadata, int bodySize, string contentType)
             {
                 var bodyId = headers.MessageId();
+                var bodyUrl = string.Format(BodyUrlFormatString, bodyId);
                 var isBinary = contentType.Contains("binary");
```

Comment:
I also saw this... plenty of other content types that are binary that do not have "binary" in the content type.

```suggestion
                var isBinary = !contentType.Contains("text") && !contentType.Contains("xml")  && !contentType.Contains("json")
```

What do you think?

Reply:
That would have to be an OR condition right? Plus I think it is dangerous to introduce this now since we could potentially be breaking things.

Reply:
@danielmarbach Not text, and not xml, and not JSON, then its binary. Could also invert the whole and do
```suggestion
                var isBinary = !(contentType.Contains("text") || contentType.Contains("xml")  || contentType.Contains("json"));
```
I'm not sure why its there. I assume that we store non binaries bodies in the message for full-text-search reasons? Storing binary messages in the message does not make sense for full-text-search. 

But.. that is just a guess.

Reply:
That is correct. The in-document storage was only ever to support full text indexing. If the tokenizer can't read it, then indexing it is pointless. In reality we only have a very small handful of officially supported serializers. We could turn this into configuration and have a list of "indexable" content types, defaulted to a known list. That might also make it easier to turn off full-text indexing. Clear out your list of indexable content types and we'll stop putting message bodies in the metadata.

Reply:
This is separate concern from this PR. We can address this is a different PR.